src/apps/competitions/utils.py

```diff
@@ -0,0 +1,49 @@
+import requests
+import logging
+
+from io import BytesIO
+
+from PIL import Image
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.files.base import ContentFile
+
+from competitions.models import Competition
+
+logger = logging.getLogger(__name__)
+
+
+def competition_download_image(competition_pk):
+    try:
```

Comment:
can we just do the lookups up here and not wrap the whole thing in a `try .. except`?